benchmark/benchmark_gemm_split_k_fp8_e4m3.py

```diff
@@ -0,0 +1,184 @@
+import os
+
+import torch
+import triton
+from utils import _print_speed_banner, _test_memory, get_current_file_directory
+
+from liger_kernel.ops.experimental.gemm_split_k_fp8_e4m3 import (
+    LigerFP8GemmSplitKFunction,
+)
+
+# enable TensorFloat32 tensor cores for better performance in benchmark
+torch.set_float32_matmul_precision("high")
+
+
+@triton.testing.perf_report(
+    [
+        triton.testing.Benchmark(
+            x_names=["m", "k", "n"],
+            x_vals=[
+                (64, 64, 64),
+                (256, 256, 256),
+                (512, 512, 512),
+                (1024, 1024, 1024),
+                (64, 128, 64),
+                (256, 512, 256),
+                (512, 1024, 512),
+                (1024, 2048, 1024),
+            ],
+            xlabel="Matrix Size (m x k x n)",
+            line_arg="provider",
+            line_vals=["liger", "torch", "torch_compile"],
+            line_names=["Liger", "PyTorch", "Torch Compile"],
+            styles=[("blue", "solid"), ("orange", "solid"), ("green", "solid")],
+            ylabel="time (ms)",
+            plot_name="gemm-split-k-fp8-fwd-speed-benchmark",
+            args={"mode": "forward", "dtype": torch.float32},
+        ),
+        triton.testing.Benchmark(
+            x_names=["m", "k", "n"],
+            x_vals=[
+                (64, 64, 64),
+                (256, 256, 256),
+                (512, 512, 512),
+                (1024, 1024, 1024),
+                (64, 128, 64),
+                (256, 512, 256),
+                (512, 1024, 512),
+                (1024, 2048, 1024),
+            ],
+            xlabel="Matrix Size (m x k x n)",
+            line_arg="provider",
+            line_vals=["liger", "torch", "torch_compile"],
+            line_names=["Liger", "PyTorch", "Torch Compile"],
+            styles=[("blue", "solid"), ("orange", "solid"), ("green", "solid")],
+            ylabel="time (ms)",
+            plot_name="gemm-split-k-fp8-full-speed-benchmark",
+            args={"mode": "full", "dtype": torch.float32},
+        ),
+    ]
+)
+def bench_speed_gemm_split_k_fp8(m, k, n, provider, mode, dtype, device="cuda"):
+    a_fp8 = torch.randn((m, k), device=device, dtype=dtype).to(torch.float8_e4m3fn)
+    b_fp8 = torch.randn((k, n), device=device, dtype=dtype).to(torch.float8_e4m3fn)
+
+    a_float = a_fp8.float().requires_grad_()
+    b_float = b_fp8.float().requires_grad_()
+
+    def fwd_liger():
+        return LigerFP8GemmSplitKFunction.apply(a_fp8, b_fp8)
+
+    def fwd_torch():
+        return torch.matmul(a_float, b_float)
```

Comment:
Computing the fp8 kernel with torch matmul on fp32 is not a quite fair comparison. A better comparison would be to compare with torch._scaled_mm with fp8 matmul such as the example here: https://gist.github.com/malfet/7874d96b99670c3da83cbb779ab770c6